src/lib.rs

```diff
@@ -0,0 +1,127 @@
+#[deny(clippy::all)]
+///! Graph API Traits
+///
+use rand::SeedableRng;
+
+#[allow(dead_code)]
+/// A graph layer name.
+struct Layer(&'static str);
```

Comment:
I do understand this might be just for simplification purposes, but are we sure that a `Layer` has a `&'static str` internally? If we give users the ability to create more layers, this will need to be a dynamic reference and thus require a lifetime? (At which point perhaps the Layer owns the name, so `std::String` might be more fitting?

Reply:
There aren't really any plans for users to be able to create graphs for now though. The Graph API is strictly internal.

Reply:
I see, gotcha!